lusSTR/scripts/marker.py

```diff
@@ -377,6 +382,11 @@ def convert(self):
         else:
             for m in re.finditer("GGGCTGCCTA", self.uas_sequence):
                 break_point = m.end()
+            try:
+                break_point
+            except NameError:
+                for m in re.finditer("TTTT", self.uas_sequence):
+                    break_point = m.end() + 10
```

Comment:
This use of the try/except mechanism is pretty unconventional.  A much more common (and IMHO, clearer) pattern would be like this.

```python
for ...:
    # found it
    break
else:
    # handle the no break case
```

In concrete terms:

```python
for m in re.finditer("GGGCTGCCTA", self.uas_sequence):
    break_point = m.end()
    break
else:
    for m in re.finditer("TTTT", self.uas_sequence):
        break_point = m.end() + 10
```

But I think we could probably improve this code even more for clarity of intent. You might consider something like this.

```python
if "GGGCTGCCTA" in self.uas_sequence:
    break_point = self.uas_sequence.index("GGGCTGCCTA") + 10
else:
    break_point = self.uas_sequence.index("TTTT") + 14
```

I'm not 100% sure about those 10 and 14 offsets, but I hope you get the idea.

Reply:
Makes sense!